iree/compiler/Codegen/LLVMCPU/LLVMCPUTileFuseAndVectorizeLinalgTensorOps.cpp

```diff
@@ -255,6 +254,23 @@ void LLVMCPUTileFuseAndVectorizePass::runOnOperation() {
     });
   }
 
+  // Apply vector unroll
+  {
+    RewritePatternSet vectorUnrollPatterns(context);
+    vector::populateVectorUnrollPatterns(
+        vectorUnrollPatterns, vector::UnrollVectorOptions().setNativeShape(
+                                  config.getNativeVectorSizeVals()));
```

Comment:
Note that this will blindly apply the same shape target for all ops independently of their rank  and/or permutation maps. This is most likely not what you want. I'm guessing doing unrolling like that is not going to help a lot. This is not a blocker in my opinion though. Just something to keep in mind.

Reply:
Are you sure about that. AFAICS, this would use the `config` on that op. So every op would see a different native vector size?